src/tools/perf/lib/libperf.c

```diff
@@ -239,15 +243,23 @@ void ucx_perf_test_start_clock(ucx_perf_context_t *perf)
     perf->current.time_acc = perf->start_time_acc;
 }
 
-static void ucx_perf_test_reset(ucx_perf_context_t *perf,
-                                ucx_perf_params_t *params)
+/* Initialize/reset all parameters that could be modified by the warm-up run */
+static void ucx_perf_test_prepare_new_run(ucx_perf_context_t *perf,
+                                          ucx_perf_params_t *params)
 {
     unsigned i;
 
-    perf->params            = *params;
     perf->max_iter          = (perf->params.max_iter == 0) ? UINT64_MAX :
                                perf->params.max_iter;
+
     perf->report_interval   = ucs_time_from_sec(perf->params.report_interval);
+
+#if _OPENMP
+    /* Suppress all other threads' output */
+    if (omp_get_thread_num() != 0)
```

Comment:
maybe better add this check inside ucx_perf_update()?

Reply:
Will do.